pkg/adapter/apiserver/stats_reporter.go

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package apiserver
+
+import (
+	"context"
+	"strconv"
+
+	. "knative.dev/eventing/pkg/metrics/metricskey"
+	"knative.dev/eventing/pkg/utils"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"knative.dev/pkg/metrics"
+	"knative.dev/pkg/metrics/metricskey"
+)
+
+var (
+	// eventCountM is a counter which records the number of events sent
+	// by an Importer.
+	eventCountM = stats.Int64(
+		"event_count",
+		"Number of events created",
+		stats.UnitDimensionless,
+	)
+)
+
+type ReportArgs struct {
+	ns                string
+	eventType         string
+	eventSource       string
+	apiServerImporter string
+}
+
+const (
+	importerResourceGroupValue = "apiserversources.sources.eventing.knative.dev"
+)
+
+// StatsReporter defines the interface for sending filter metrics.
+type StatsReporter interface {
+	ReportEventCount(args *ReportArgs, responseCode int) error
+}
+
+var _ StatsReporter = (*reporter)(nil)
+
+// reporter holds cached metric objects to report filter metrics.
+type reporter struct {
+	namespaceTagKey             tag.Key
+	eventTypeTagKey             tag.Key
+	eventSourceTagKey           tag.Key
+	importerNameTagKey          tag.Key
+	importerResourceGroupTagKey tag.Key
+	responseCodeKey             tag.Key
+	responseCodeClassKey        tag.Key
+	filterResultKey             tag.Key
```

Comment:
remove `filterResultKey`, that is just for triggers

Reply:
removed